diff -u a/src/ucs2.h b/src/ucs2.h
--- a/src/ucs2.h	2020-05-20 12:36:19.089977883 -0400
+++ b/src/ucs2.h	2020-05-20 12:36:30.938042176 -0400
@@ -26,12 +26,11 @@
 ucs2len(const void *vs, ssize_t limit)
 {
 	ssize_t i;
-	const uint16_t *s = vs;
 	const uint8_t *s8 = vs;
 
 	for (i = 0;
-	     i < (limit >= 0 ? limit : i+1) && s8[0] != 0 && s8[1] != 0;
-	     i++, s8 += 2, s++)
+	     i < (limit >= 0 ? limit : i+1) && !(s8[0] == 0 && s8[1] == 0);
+	     i++, s8 += 2)
 		;
 	return i;
 }
